tests/provider: Ensure misspell runs on .changelog/ files before merge

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

We currently have GitHub Actions monitoring `CHANGELOG.md` for spelling issues when that file is changed and on the main branch. We can now catch them before they are merged via changes to the `.changelog/` directory in contributions. This will prevent situations like this on unrelated contributions:

```
CHANGELOG.md:25:58: "perserve" is a misspelling of "preserve"
Error: Process completed with exit code 2.
```

Might be as simple as adjusting the existing CHANGELOG workflow to also look at that directory.

### References

* Further CHANGELOG contents checking: https://github.com/hashicorp/terraform-provider-aws/issues/17135

--- a/.github/workflows/changelog.yml
+++ b/.github/workflows/changelog.yml
@@ -6,6 +6,7 @@ on:
       - 'release/**'
   pull_request:
     paths:
+      - .changelog/
       - .go-version
       - CHANGELOG.md
   pull_request_target:
@@ -70,4 +71,4 @@ jobs:
           path: ~/go/pkg/mod
           key: ${{ runner.os }}-go-pkg-mod-${{ hashFiles('go.sum') }}
       - run: cd tools && go install github.com/client9/misspell/cmd/misspell
-      - run: misspell -error -source text CHANGELOG.md
+      - run: misspell -error -source text CHANGELOG.md .changelog

--- a/.github/workflows/changelog.yml
+++ b/.github/workflows/changelog.yml
@@ -6,7 +6,7 @@ on:
       - 'release/**'
   pull_request:
     paths:
-      - .changelog/
+      - .changelog/*
       - .go-version
       - CHANGELOG.md
   pull_request_target:

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -81,7 +81,7 @@ docscheck:
 		-allowed-resource-subcategories-file website/allowed-subcategories.txt \
 		-ignore-side-navigation-data-sources aws_alb,aws_alb_listener,aws_alb_target_group,aws_kms_secret \
 		-require-resource-subcategory
-	@misspell -error -source text CHANGELOG.md
+	@misspell -error -source text CHANGELOG.md .changelog
 
 lint: golangci-lint awsproviderlint importlint
 

